pageserver/src/tenant/timeline/compaction.rs

```diff
@@ -47,10 +47,14 @@ impl Timeline {
     /// TODO: cancellation
     pub(crate) async fn compact_legacy(
         self: &Arc<Self>,
-        _cancel: &CancellationToken,
+        cancel: &CancellationToken,
         flags: EnumSet<CompactFlags>,
         ctx: &RequestContext,
     ) -> Result<(), CompactionError> {
+        if flags.contains(CompactFlags::EnhancedGcBottomMostCompaction) {
+            return self.compact_with_gc(cancel, ctx).await;
+        }
```

Comment:
Hm, the flags aren't really meant for "modes".

I don't care too stronly about this for now, but, is there any functional reason why we don't treat bottommost compaction as another _step_ instead of returning early?

Reply:
I feel for now it makes more sense to only run bottom-most compaction when the compaction API gets called, so that we can know the consequences of the compaction without interference from other parts of the system.